scripts/quests/sandoria/A_Timely_Visit.lua

```diff
@@ -0,0 +1,322 @@
+            },
+        },
+    },
+
+    -- NM fight to end
+    {
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and
+                vars.Stage == 1
+        end,
+
+        [xi.zone.JUGNER_FOREST] =
+        {
+            ['qm1'] =
+            {
+                onTrigger = function(player, npc)
+                    local hour = VanadielHour()
+                    local progress = quest:getVar(player, 'Prog')
+                    local mobsSpawned = (GetMobByID(jugnerID.mob.GIOLLEMITTE):isSpawned() or GetMobByID(jugnerID.mob.GIOLLEMITTE + 1):isSpawned())
+                    local waitOver = os.time() >= npc:getLocalVar('Wait')
+                    local isDaytime = (hour >= 6 and hour <= 17)
+
+                    if progress == 6 and not mobsSpawned then
+                        return quest:progressEvent(18)
+                    elseif progress == 5 and waitOver and not mobsSpawned then
+                        if not isDaytime then
+                            for i = 0, 1 do
+                                SpawnMob(jugnerID.mob.GIOLLEMITTE + i):updateClaim(player)
+                            end
+
+                            return quest:messageSpecial(jugnerID.text.SENSE_OF_FOREBODING)
+                        else
+                            return quest:messageSpecial(jugnerID.text.UNABLE_TO_INVESTIGATE)
+                        end
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+
+                [18] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 7)
+                end,
+            },
+
+            -- Player can kill EITHER NM to progress
+            ['Giollemitte_B_Feroun'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if
+                        mob:getID() == jugnerID.mob.GIOLLEMITTE and
+                        quest:getVar(player, 'Prog') == 5
+                    then
+                        quest:setVar(player, 'Prog', 6)
+                    end
+                end,
+            },
+
+            -- Player can kill EITHER NM to progress
+            ['Skeleton_Esquire_NM'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if
+                        mob:getID() == jugnerID.mob.GIOLLEMITTE + 1 and
+                        quest:getVar(player, 'Prog') == 5
+                    then
+                        quest:setVar(player, 'Prog', 6)
+                    end
+                end,
+            },
+
+            -- Resets the progress if the player killed one of the NMs but didn't check the ??? before zoning. Will result in the player needing to refight.
+            onZoneOut = function(player)
+                if quest:getVar(player, 'Prog') == 6 then
+                    quest:setVar(player, 'Prog', 5)
+                end
+            end,
```

Comment:
Instead, why not set a local variable on players as "killedNM" when killing one of the NMs?
That would handle deleting the variable directly